pkg/docker/local_builder.go

```diff
@@ -233,19 +231,15 @@ func buildPipe(pipe io.ReadCloser, logWriter logger.Logger) (lastLogs []string,
 
 	scanner := bufio.NewScanner(pipe)
 	currentSection := SectionStartingBuild
-	currentLogLines := []string{}
-	logWriter.Infof("  * %s", currentSection)
+	logWriter.Infof(currentSection)
 
 	for scanner.Scan() {
 		line := scanner.Text()
 		logWriter.Debug(line)
 
 		if strings.Contains(line, sectionPrefix) {
 			currentSection = strings.SplitN(line, sectionPrefix, 2)[1]
-			currentLogLines = []string{}
-			logWriter.Infof("  * %s", currentSection)
-		} else {
-			currentLogLines = append(currentLogLines, line)
+			logWriter.Infof(currentSection)
```

Comment:
How does this display stack traces if some build step fails? The previous behavior made it quite clear what exactly failed. Are we now only showing the last _n_ lines of the error message? Is there a way to see the full failure trace from docker build?

Reply:
Yes -- behavior is the same but this is now handled by the consumer of the log rather than the writer.